include/manager.hpp

```diff
@@ -30,15 +31,110 @@ class Manager
+     *
+     * @param[in] i_input Input to write keyword.
+     *
+     * @return Data which is written successfully.
+     */
+    types::ReturnValType
+        writeKeywordOnSource(const types::WriteKeywordSignature i_input);
+
+    /**
+     * @brief Write on BMC cache memory
+     * API can be used to perform keyword writes on BMC cache (On D-bus).
+     *
+     * @param[in] i_input - Input to write keyword.
+     *
+     * @return Data which is written successfully.
+     */
+    types::ReturnValType
+        writeKeywordOnCache(const types::WriteKeywordSignature i_input);
+
+    /**
+     * @brief Read keyword from hardware
+     * API can be used to perform keyword reads directly from EEPROM.
+     *
+     * @param[in] i_input - Input to read keyword value.
+     *
+     * @return Read value.
+     */
+    types::ReturnValType
+        readKeywordFromSource(const types::ReadKeywordSignature i_input);
+
+    /**
+     * *@brief Read keyword from cache
+     * API can be used to perform keyword reads directly from BMC cache memory
+     * (D-bus)
+     *
+     * @param[in] i_input - Input to read keyword value.
+     *
+     * @return Read value.
+     */
+    types::ReturnValType
+        readKeywordFromCache(const types::ReadKeywordSignature i_input);
+
+    /**
+     * @brief Collect single FRU VPD
+     * API can be used to perform VPD collection for the given FRU.
+     *
+     * @param[in] i_invPath - Inventory path
+     */
+    void collectSingleFruVpd(const sdbusplus::message::object_path& i_invPath);
+
+    /**
+     * @brief Delete single FRU VPD
+     * API can be used to perform VPD deletion for the given FRU.
+     *
+     * @param[in] i_invPath - Inventory path
+     */
+    void deleteSingleFruVpd(const sdbusplus::message::object_path& i_invPath);
+
+    /**
+     * @brief Read Location code
+     * API can be used to read location code for the given FRU inventory path.
+     *
+     * @param[in] i_invPath - Inventory path
+     * @param[in] i_version - Expanded version(0x00); Unexpanded version(0x01)
+     *
+     * @return Based on the requested version, return either expanded or
+     * unexpanded location code. By default return unexpanded location code.
+     */
+    std::string
+        readLocationCode(const sdbusplus::message::object_path& i_invPath,
+                         const uint8_t i_version = 0x01);
+
+    /**
+     * @brief Get Hardware path
+     * API can be used to get hardware path for the given inventory path.
+     *
+     * @param[in] i_invPath - Inventory path
+     *
+     * @return Corresponding hardware path.
+     */
+    std::string getHwPath(const sdbusplus::message::object_path& i_invPath);
+
+    /**
+     * @brief Is VPD collection done
+     * API can be used to check if all FRU VPDs are collected or not.
+     *
+     * @return true if all FRU VPDs are collected successfully, false otherwise.
+     */
+    bool isVpdCollectionDone();
```

Comment:
Don't need return type. Value to be updated on Dbus.

Reply:
how the stakeholders will get the value without having return type?

Reply:
As discussed, it is a setter method. Make it private and remove return type.